core/includes/config.inc

```diff
@@ -513,6 +632,39 @@ class Config {
   }
 
   /**
+   * Merge new defaults with this configuration object and update the default
+   * hash signature.  This method will not overwrite existing configuration
+   * settings.  It only adds settings that are missing.
+   *
+   * @param array $data
+   *   The new default configuration data.
+   *
+   * @return Config
+   *   The configuration object.
+   */
+  public function mergeDefaultData(array $data) {
+    $this->replaceData($this->data + $data);
+    $this->setDefaultHash($data);
+    // A load would destroy the data just set (for example on import).
+    $this->isLoaded = TRUE;
+    return $this;
+  }
+
+  /**
+   * Computes the hash signature of the given config data or the current config
+   * data and save it in the _default_hashes section.
+   *
+   * @param array $data
+   *   The data to hash.  Uses the current configuration data set if NULL.
+   */
+  public function setDefaultHash($data=NULL) {
+    if ($data === NULL) {
+      $data = $this->data;
+    }
+    $algo = self::DEFAULT_HASH_ALGORITHM;
+    $this->data['_default_hashes'][$algo] = Config::Hash($data, $algo);
```

Comment:
I'd prefer it.  The keyed array makes it obvious what algorithm we're using, and if BD core or any site ever has to use something else (local laws, sha256 becomes "broken", performance, whatever), the upgrade path is easy and clear.